engines/e_padlock.c

```diff
@@ -88,16 +100,19 @@ static int padlock_bind_helper(ENGINE *e)
 
     /* Generate a nice engine name with available features */
     BIO_snprintf(padlock_name, sizeof(padlock_name),
-                 "VIA PadLock (%s, %s)",
+                 "VIA PadLock (%s, %s, %s)",
                  padlock_use_rng ? "RNG" : "no-RNG",
-                 padlock_use_ace ? "ACE" : "no-ACE");
+                 padlock_use_ace ? "ACE" : "no-ACE",
+                 padlock_use_ccs ? "CCS" : "no-CCS");
 
     /* Register everything or return with an error */
     if (!ENGINE_set_id(e, padlock_id) ||
         !ENGINE_set_name(e, padlock_name) ||
         !ENGINE_set_init_function(e, padlock_init) ||
         (padlock_use_ace && !ENGINE_set_ciphers(e, padlock_ciphers)) ||
-        (padlock_use_rng && !ENGINE_set_RAND(e, &padlock_rand))) {
+        (padlock_use_rng && !ENGINE_set_RAND(e, &padlock_rand)) ||
+        (padlock_use_ccs && !ENGINE_set_ciphers(e, zx_ciphers)) ||
```

Comment:
Note: ENGINE_set_ciphers overwrited the AES ciphers in padlock_ciphers
so you must merge the available ciphers and use this API only once.
Have you tested the AES ciphers still work?

Reply:
In fact，zx_ciphers have included all padlock_ciphers. 
And padlock_ciphers are the parts of zx_ciphers.